### PR TITLE
fix(buildDockerAndPublishImage.groovy): fix ath build

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkinsci/acceptance-test-harness/pull/1356#issuecomment-1706939223
and following https://github.com/jenkins-infra/pipeline-library/pull/739

this should correct the bug in the Path due to the absolute vs relative definition

tested on wiki (default path): https://infra.ci.jenkins.io/job/docker-jobs/job/docker-confluence-data/job/main/131/pipeline-graph/
tested on ATH (manually defined path): https://infra.ci.jenkins.io/job/docker-jobs/job/acceptance-test-harness/job/master/289/pipeline-graph/
tested on docker-inbound-agent (windows/legacy build): https://infra.ci.jenkins.io/job/docker-jobs/job/docker-inbound-agents/job/main/178/pipeline-console/

### CODE REVIEW

File: resources/io/jenkins/infra/docker/Makefile [hunk 1]
```diff
@@ -22,6 +29,16 @@ GIT_SCM_URL ?= $(shell git config --get remote.origin.url)
...
 
+export 	IMAGE_DEPLOY_NAME 		\
+		TAG_NAME 				\
+		BAKE_TARGETPLATFORMS 	\
+		IMAGE_DOCKERFILE 		\
+		IMAGE_DIR 				\
+		GIT_COMMIT_REV 			\
+		GIT_SCM_URL 			\
+		BUILD_DATE 				\
+		SCM_URI
```
Comment (dduportal) [inline resources/io/jenkins/infra/docker/Makefile:40]:
```suggestion
export 	IMAGE_DEPLOY_NAME \
		TAG_NAME \
		BAKE_TARGETPLATFORMS \
		IMAGE_DOCKERFILE \
		IMAGE_DIR \
		GIT_COMMIT_REV \
		GIT_SCM_URL \
		BUILD_DATE \
		SCM_URI
```

nitpick: formatting

**Review by dduportal — CHANGES_REQUESTED** [review]
Not sure yet where did it failed but the commit https://github.com/jenkins-infra/pipeline-library/pull/745/commits/acf5a061d9c1e7df7e079dcb592718ab84d14977 broke the windows build: https://infra.ci.jenkins.io/job/docker-jobs/job/docker-inbound-agents/job/main/178/

But the ATH build is fixed by this change 👍

**Comment by smerle33** [comment]
> Not sure yet where did it failed but the commit [acf5a06](https://github.com/jenkins-infra/pipeline-library/commit/acf5a061d9c1e7df7e079dcb592718ab84d14977) broke the windows build: https://infra.ci.jenkins.io/job/docker-jobs/job/docker-inbound-agents/job/main/178/
> 
> But the ATH build is fixed by this change 👍

seem like linting with absolute path is not working  : 

```
== Linting C:\maven\jdk11\Dockerfile.nanoserver with hadolint:
hadolint --format=json "C:\maven\jdk11\Dockerfile.nanoserver" > "C:\Jenkins\workspace\_jobs_docker-inbound-agents_main\jenkinsciinfra-inbound-agent-maven-jdk11-nanoserver-hadolint-1694090862736.json"
powershell.exe : hadolint.exe: C:\maven\jdk11\Dockerfile.nanoserver: withBinaryFile: does not exist (No such file or directory)
At C:\Jenkins\workspace\_jobs_docker-inbound-agents_main@tmp\durable-891865a5\powershellWrapper.ps1:3 char:1
+ & powershell -NoProfile -NonInteractive -ExecutionPolicy Bypass -Comm ...
+ 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     + CategoryInfo          : NotSpecified: (hadolint.exe: C...e or directory):String) [], RemoteException
     + FullyQualifiedErrorId : NativeCommandError
  
make: *** [Makefile:50: lint] Error 1
```

still don't get how I missed it during my test, it happend in the build I launched myself ... thanks @dduportal for catching it !
while it was in relative ... need to check if the absolute path is correct : 
```
 == Linting maven\jdk11\Dockerfile.nanoserver with hadolint:
 hadolint --format=json "maven\jdk11\Dockerfile.nanoserver" > "C:\Jenkins\workspace\_jobs_docker-inbound-agents_main\jenkinsciinfra-inbound-agent-maven-jdk11-nanoserver-hadolint-1693582283540.json"
 == Lint succeeded
```

**Comment by smerle33** [comment]
this new version go through docker inbound agents (windows) build : https://infra.ci.jenkins.io/job/docker-jobs/job/docker-inbound-agents/job/main/182/console

lets try again the others : 

- [ ] wiki (default path): https://infra.ci.jenkins.io/job/docker-jobs/job/docker-confluence-data/job/main/131/pipeline-graph/
- [ ] ATH (manually defined path): 
- [ ] docker-inbound-agent (windows/legacy build):

**Review by dduportal — APPROVED** [review]
Looks like your fix does the job 👏

**Pull request merged**